### ISSUE TITLE
Add support for TDD

### ISSUE THREAD

**Issue by elpie89** [BabylonJS/Exporters#606]
Adding features is becoming more and more complex, we should add some sample scene and implement a proper Test-Driven development pipeline

**Comment by Drigax** [comment]
I completely agree. I think we can quickly get some test coverage via an integration testing approach (Similar to our CI pipeline for Babylon.JS):

- [ ] Itemize functionality and create individual test cases for functionalty

- [ ] Create harness for batch export of test scenes

- [ ] Create library of per-testcase validation images 

- [ ] Create harness for loading export in trusted viewer and performing per-pixel comparison with validation image

- [ ] Create simple harness for doing 2 & 4

Eventually:

- [ ] Setup Azure instance to run #5 for pull requests

**Comment by deltakosh** [comment]
Up?

**Comment by deltakosh** [comment]
Removing 4.2 and flagging it as future

**Comment by Drigax** [comment]
I definitely think this should be a long-term goal of this project. via #969 we're starting to run into more concerns about regressing behavior, and without some regression testing solution, we'll never know if our changes are breaking some known behavior.
